src/main/scala/za/co/absa/commons/version/Component.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons.version
+
+/**
+  * Represents a one component of a version.
+  * E.g. for the version string "1.foo.42" the components would be "1", "foo" and "42".
+  */
+sealed trait Component extends Ordered[Component] {
+  final override def compare(that: Component): Int = {
+    val c = comparator
+    if (c isDefinedAt that) c(that)
+    else -(that compare this)
+  }
+
+  def comparator: PartialFunction[Component, Int]
+}
+
+object Component {
+  def apply(s: String): Component =
+    if (s.forall(_.isDigit)) NumericComponent(s.toInt)
+    else StringComponent(s)
+}
+
+/**
+  * An empty component that is less that any other component as long as the other component claims the opposite.
```

Comment:
The sentence doesn't make sense to me. What is the part _...as long as the other component claims the opposite._ trying to say?
Also `An empty component that is less than any other component`

Reply:
Fixed.